lib/Service/Group/AbstractGroupService.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @author Matthias Rella <mrella@pisys.eu>
+ *
+ * Mail
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service\Group;
+
+abstract class AbstractGroupService {
+
+	/**
+	 * Search the service's groups.
+	 *
+	 * @param string $term
+	 * @return string
+	 */
+	abstract public function search($term);
+
+	/**
+	 * Get the group's namespace.
+	 *
+	 * @return string
+	 */
+	abstract public function getNamespace();
```

Comment:
Do we really need these namespaces? This kinda looks like an unnecessary generalization.

Reply:
Yes, I think so. There are two reasons:
1. When `GroupIntegration` once would be extended for Contacts groups, we need a means to tell apart whether it's a Nextcloud or a Contacts group, since both applications can have equally named groups. Eg. nextcloud:mygroup vs. contacts:mygroup.
2. We need to distinguish nextcloud groups from single recipients, since Nextcloud groups could look like email addresses. Eg. nextcloud:group@domain.com vs. group@domain.com ...